pandas/_libs/groupby.pyx

```diff
@@ -863,7 +874,15 @@ def group_ohlc(
 
     if K > 1:
         raise NotImplementedError("Argument 'values' must have only one dimension")
-    out[:] = np.nan
+
+    if int64float_t is float32_t or int64float_t is float64_t:
+        out[:] = np.nan
+    else:
+        out[:] = 0
+
+    first_element_set = np.zeros((<object>counts).shape, dtype=np.uint8)
```

Comment:
Could `N` or `K` be used to set `zeros` shape?

Reply:
No, we need the number of groups to get a corresponding mask.